Microsoft.Azure.Cosmos/src/Telemetry/VmMetadataApiHandler.cs

```diff
@@ -103,7 +103,7 @@ internal static async Task<AzureVMMetadata> ProcessResponseAsync(HttpResponseMes
         /// <returns>machine id</returns>
         internal static string GetMachineId()
         {
-            if (VmMetadataApiHandler.azMetadata != null)
+            if (VmMetadataApiHandler.azMetadata != null && VmMetadataApiHandler.azMetadata.Compute != null)
```

Comment:
Just to be on the safe side it's probably best to check all the properties. 

```suggestion
            if (VmMetadataApiHandler?.azMetadata?.Compute?.VMId)
```